src/main/scala/com/gilcloud/sbt/gitlab/GitlabPlugin.scala

```diff
@@ -107,8 +111,34 @@ object GitlabPlugin extends AutoPlugin {
         .value,
       publish := publish.dependsOn(headerAuthHandler).value,
       publishTo := (ThisProject / publishTo).value.orElse {
-        gitlabProjectId.value.map(p => "gitlab-maven" at s"https://${gitlabDomain.value}/api/v4/projects/$p/packages/maven") orElse
-        gitlabGroupId.value.map(g => "gitlab-maven" at s"https://${gitlabDomain.value}/api/v4/groups/$g/-/packages/maven")
+        gitlabProjectId.value.map(gitlabProjectRepository(gitlabDomain.value, _)) orElse
+        gitlabGroupId.value.map(gitlabGroupRepository(gitlabDomain.value, _))
+      },
+      resolvers ++= gitlabProjectId.value
```

Comment:
By only looking at the `gitlabProjectId`, it's hard to add multiple Gitlab repositories with their credentials to resolvers. I would suggest adding a method that returns the `Seq[Setting]` needed for adding a Gitlab repo

Reply:
I've experimented with defining such a method. Unfortunately, to add repository configuration it has to access the current `csrConfiguration` (both the default and `updateClassifiers`/`updateSbtClassifiers`). It doesn't look pretty: either method should accept the three `csrConfiguration`s as parameters, or return a `Task[Seq[Def.Setting[_]]]`.

Reply:
And both options don't seem to work :(

Reply:
[Adding support for multiple resolvers](https://github.com/azolotko/sbt-gitlab/pull/5) required quite a rework.

TODO: support multiple realms per GitLab domain.